testnet/stacks-node/src/tests/signer/mod.rs

```diff
@@ -323,20 +323,16 @@ impl<S: Signer<T> + Send + 'static, T: SignerEventTrait + 'static> SignerTest<Sp
         )
         .unwrap();
 
-        let t_start = Instant::now();
-        while test_observer::get_mined_nakamoto_blocks().is_empty() {
-            assert!(
-                t_start.elapsed() < timeout,
-                "Timed out while waiting for mined nakamoto block event"
-            );
-            thread::sleep(Duration::from_secs(1));
-        }
+        wait_for(timeout.as_secs(), || {
```

Comment:
This change is required because in multiple node set ups, there is only one test observer per node and you cannot guarantee which mines and therefore sees the block increment. This fixes a few tests.